ansible/roles/lib_aos_modules/library/aos_rhsm_repos.py

```diff
@@ -0,0 +1,202 @@
+                              'enabled': int(repo[3].split('Enabled:')[1].strip()) == 1
+                             })
+
+        return all_repos
+
+
+    def query_repos(self, qtype='list'):
+        """query for repos"""
+
+        cmd = [RHSM_BIN, 'repos']
+
+        if qtype == 'enabled':
+            cmd.append('--list-enabled')
+        elif qtype == 'disabled':
+            cmd.append('--list-disabled')
+        else:
+            cmd.append('--list')
+
+        rcode, out, err = self.module.run_command(cmd)
+        if rcode != 0:
+            raise RHSMError('error querying repositories. rc=[%s] cmd=[%s] err=[%s]' % (rcode, cmd, err))
+
+        if 'no available repositories matching' in out:
+            return []
+
+        return RHSMRepos.parse_repos(out)
+
+    def needs_update(self):
+        """verify that the enabled repos are enabled"""
+        repos = self.query_repos(qtype='enabled')
+        if len(repos) != len(self.module.params['enabled']):
+            return True, repos
+
+        # build sets of the repo ids and compare them
+        rids = set([rep['id'] for rep in repos])
+        ren = set(self.enabled)
+        if rids - ren or ren - rids:
+            return True, repos
+
+        return False, repos
+
+    def check_identity(self):
+        """verify that this system is entitled"""
+        rcode, _, _ = self.module.run_command([RHSM_BIN, 'identity'])
+        if rcode != 0:
+            self.module.fail_json(msg="command `subscription-manager identity` failed.  exit code non zero.")
+
+    @staticmethod
+    def run_ansible(module):
+        """run the ansible code"""
+        rhsm = RHSMRepos(module)
+
+        # step 1: verify we are entitled
+        rhsm.check_identity()
+
+        # step 2: if we are state=list, query and return results
+        if rhsm.module.params['state'] == 'list':
+            repos = rhsm.query_repos(qtype=module.params['query'])
+
+            rhsm.module.exit_json(changed=False, repos=repos)
+
+        # step 2: if we are state=present, check if we need to update
+        elif module.params['state'] == 'present':
+            update, repos = rhsm.needs_update()
+            if not update:
+                module.exit_json(changed=update, repos=repos)
+
+            # step 3: we need to set our repositories
+            rhsm.set_repositories()
+            repos = rhsm.query_repos(qtype='enabled')
+            module.exit_json(changed=update, repos=repos)
+
+        module.fail_json(msg="unsupported state.")
+
+def main():
+    """Create the ansible module and run the ansible code"""
+    module = AnsibleModule(
+        argument_spec=dict(
+            enabled=dict(default=None, type='list'),
+            disabled=dict(default=None, type='list'),
+            state=dict(default='present', choices=['list', 'present']),
+            query=dict(default='all', choices=['all', 'enabled', 'disabled']),
+        ),
+        supports_check_mode=True
```

Comment:
How is `check_mode` being supported in this module?

I probably missed it, but I looked through the code twice for it and don't see anything that's checking for check_mode.

Is this maybe an automatic feature of `module.run_command`?

The [docs](http://docs.ansible.com/ansible/dev_guide/developing_modules.html#check-mode) say you need to do an if statement using `module.check_mode`.